packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js

```diff
@@ -309,12 +292,13 @@ export const BlocksDropdown = ({ disabled }) => {
   React.useEffect(() => {
     if (editor.selection) {
       const [anchorNode] = Editor.parent(editor, editor.selection.anchor); // Get the parent node of the anchor
+      const anchorBlockKey = Object.keys(blocks).find((blockKey) =>
+        blocks[blockKey].matchNode(anchorNode)
+      );
 
       if (anchorNode) {
-        const blockKey = getAnchorBlockKey(blocks, anchorNode);
-
-        if (blockKey && blockKey !== blockSelected) {
-          setBlockSelected(blockKey);
+        if (anchorBlockKey && anchorBlockKey !== blockSelected) {
+          setBlockSelected(anchorBlockKey);
```

Comment:
you can flatten these 2 into 1 now

Reply:
I changed the implementation, I don't use anymore the getAnchorBlockKey